### ISSUE TITLE
Camera streams poor performance

### ISSUE THREAD

**Issue by Lumia1r** [homebridge-simplisafe3/homebridge-simplisafe3#204]
I’m running latest versions in docker container on synology 418play.
I have 2 nest cams and 2 Netatmo HomeKit cameras all working fine.

As soon as I enable camera support on the SimpliSafe plugin I see greatly reduced performance in all my camera streams, often timing out entirely. 

Frequent messages in Homebridge about my SimpliSafe cameras being slow to respond.

Any ideas or help would be hugely appreciated

**Comment by shamoon** [comment]
Hi @Lumia1r , are you saying your other streams get slower when cameras are enabled on this plug-in? Camera support is tough because unlike other cameras we have to tap into a stream via the web (so camera —> internet —> plug-in) whereas others can stream directly from the cameras (not possible with SS). As a result streaming can be somewhat resource intensive.

_But_ that should only be when you’re actually streaming video, when it’s not streaming it should have no effect except when polling for snapshots (which in iOS/macOS is only periodically when the Home.app is open)

**Comment by Lumia1r** [comment]
> Hi @Lumia1r , are you saying your other streams get slower when cameras are enabled on this plug-in? Camera support is tough because unlike other cameras we have to tap into a stream via the web (so camera —> internet —> plug-in) whereas others can stream directly from the cameras (not possible with SS). As a result streaming can be somewhat resource intensive.
> 
> _But_ that should only be when you’re actually streaming video, when it’s not streaming it should have no effect except when polling for snapshots (which in iOS/macOS is only periodically when the Home.app is open)

This was the report yes, however after disabling camera support and re-installing my nest plugin, then re-enabling camera support.. for now things are working normally. Perhaps it was caused by something else.

I am seeing "The image snapshot handler for the given accessory is slow to respond!" a lot for both of my cameras though.

Thank you

**Comment by shamoon** [comment]
Yes, those are warnings from Homebridge that are the result of the slow round-trip snapshots take via the internet. All you can really do is ensure your cameras have the best network performance as possible and from there we're at the mercy of SimpliSafe's disappointingly slow software / servers. Glad things are working.
